core/workspace_svg.js

```diff
@@ -1467,40 +1467,56 @@ Blockly.WorkspaceSvg.prototype.highlightBlock = function(id, opt_state) {
 };
 
 /**
- * Paste the provided block onto the workspace.
- * @param {!Element|!DocumentFragment} xmlBlock XML block element or an empty
- *     DocumentFragment if the block was an insertion marker.
- */
-Blockly.WorkspaceSvg.prototype.paste = function(xmlBlock) {
-  if (!this.rendered || !xmlBlock.tagName || xmlBlock.getElementsByTagName('block').length >=
-      this.remainingCapacity()) {
+ * Pastes the provided block or workspace comment onto the workspace.
+ * Does not check whether there is remaining capacity for the object, that
+ * should be done before calling this method.
+ * @param {!Object|!Element|!DocumentFragment} state The representation of the
+ *     thing to paste.
+ */
+Blockly.WorkspaceSvg.prototype.paste = function(state) {
+  if (!this.rendered || !state['type'] && !state.tagName) {
     return;
   }
-  // The check above for tagName rules out the possibility of this being a DocumentFragment.
-  xmlBlock = /** @type {!Element} */ (xmlBlock);
   if (this.currentGesture_) {
     this.currentGesture_.cancel();  // Dragging while pasting?  No.
   }
-  if (xmlBlock.tagName.toLowerCase() == 'comment') {
-    this.pasteWorkspaceComment_(xmlBlock);
+  if (state['type']) {
```

Comment:
I think this is just to check if we are in the jso system? If that is the case, could you add a comment explaining this above the if statement?